Makefile

```diff
@@ -3180,6 +3180,7 @@ cocciclean:
 clean: profile-clean coverage-clean cocciclean
 	$(RM) *.res
 	$(RM) $(OBJECTS)
+	$(RM) headless-git.o
```

Comment:
Is there a reason `headless-git.o` is not included in `$(OBJECTS)`? What's so special about `headless-git.o` that it get's special cased here and later in the third fixup, just to give it the same treatment as the other `.o` files?

Reply:
It is not intended to link into the typical `git.exe`, but other than that I'm not sure.

Reply:
I thought about putting it in $(OBJECTS) but decided it was an unnecessary risk just to avoid adding another cleanup line.  It is referenced by many things, such as building up the dependencies, the implicit compiler rules, and various $(filter) expressions and I was wanting to avoid any unintended side-effects.  This particular file has a unique compilation (in WinMain mode), so it needs to be broken out anyway, so I didn't see the need to glob it in with the others.

Reply:
Ok, that makes sense.

Reply:
Indeed, the rules would make it so that it is compiled in the same way as all the other `.c` files, which is _very much not_ what we want with `headless-git.exe`. It is intended to be a super small, standalone `.exe` that is compiled in GUI mode so that it does not pop up a command-line window when run from the Scheduled Tasks.